builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -334,7 +334,26 @@ func resourceAwsSecurityGroupUpdate(d *schema.ResourceData, meta interface{}) er
 		}
 	}
 
-	if err := setTags(conn, d); err != nil {
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+		log.Printf("[INFO] Setting tags SecurityGroup ID: %s", d.Id())
+		err = setTags(conn, d)
```

Comment:
Is it possible we could add this logic to the `setTags` method?

- https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/tags.go#L22

It seems reasonable that we could check the error code for the `NotFound` text and issue the retry